pkg/pointerdb/pointerdb.go

```diff
@@ -131,136 +129,48 @@ func (s *Server) Get(ctx context.Context, req *pb.GetRequest) (resp *pb.GetRespo
 	}, nil
 }
 
-// List calls the bolt client's List function and returns all Path keys in the Pointers bucket
+// List returns all Path keys in the Pointers bucket
 func (s *Server) List(ctx context.Context, req *pb.ListRequest) (resp *pb.ListResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 	s.logger.Debug("entering pointerdb list")
 
-	limit := int(req.GetLimit())
-	if limit <= 0 || limit > ListPageLimit {
-		limit = ListPageLimit
-	}
-
-	if err = s.validateAuth(req.GetAPIKey()); err != nil {
+	if err = s.validateAuth(req.APIKey); err != nil {
+		s.logger.Error("authentication failed", zap.Error(err))
 		return nil, err
 	}
 
-	prefix := paths.New(req.GetPrefix())
-
-	// TODO(kaloyan): here we query the DB without limit. We must optimize it!
-	keys, err := s.DB.List([]byte(req.GetPrefix()+"/"), 0)
+	rawItems, more, err := storage.ListV2(s.DB, storage.ListOptions{
+		Prefix:       storage.Key(req.Prefix),
+		StartAfter:   storage.Key(req.StartAfter),
+		EndBefore:    storage.Key(req.EndBefore),
+		Recursive:    req.Recursive,
+		Limit:        storage.Limit(req.Limit),
+		IncludeValue: true,
```

Comment:
We need the values only if `req.MetaFlags != meta.None`. Otherwise, it would be faster to just query the keys (object paths).